modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/config/DefaultConfigurationInstanceProviderFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.config;
+
+import static org.mule.module.extension.internal.util.MuleExtensionUtils.getInitialiserEvent;
+import org.mule.api.MuleContext;
+import org.mule.extension.ExtensionManager;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.runtime.ConfigurationInstanceProvider;
+import org.mule.module.extension.internal.manager.ExtensionManagerAdapter;
+import org.mule.module.extension.internal.runtime.DynamicConfigPolicy;
+import org.mule.module.extension.internal.runtime.resolver.ResolverSet;
+
+/**
+ * Default implementation of {@link ConfigurationInstanceProviderFactory}
+ *
+ * @since 4.0
+ */
+public final class DefaultConfigurationInstanceProviderFactory implements ConfigurationInstanceProviderFactory
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> ConfigurationInstanceProvider<T> createConfigurationInstanceProvider(
```

Comment:
that depends on your point of view. From the caller's point of view, I just want a ConfigurationInstanceProvider and I'd like to be decoupled from the fact that those can be static or dynamic. To return one or other type can be though of as part of the creation logic that this factory is trying to extract.